com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/ShaderPass.template

```diff
@@ -31,7 +32,22 @@ Pass
     #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl"
     #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Texture.hlsl"
     #include "Packages/com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariables.hlsl"
+
+	$features.graphVFX: #define HAVE_VFX_MODIFICATION
+
+    // [NOTE-FRAG-INPUTS]
+    // Omit the HDRP FragInput struct definition since VFX provides an arbitrary number of interpolators
+    // (this is because properties are evaluated by the graph on the fly in the vertex stage)
+    // These interpolates must be transfered into the FragInputs so they are available for the Surface evaluation.
+    // Thus we procedurally generate a FragInput struct.
+    // Warning: Omitting this means we no longer define the GetVaryingsDataDebug..
+    #ifndef HAVE_VFX_MODIFICATION
     #include "Packages/com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/FragInputs.hlsl"
+    #else
+    // For now, move the full struct splice to this point (contains FragInputs).
+    $splice(PassStructs)
```

Comment:
Not very happy with how I currently move the entire struct splice up to this point, eager to hear thoughts about it.  This also might have unforeseen issues that will later be caught in the CI. 

Reminder: we must do this because FragInputs contains all of the incoming interpolator that got configured by VFX. Thus we must override FragInputs and define all of the structs here so that anything dependent on FragInputs can still get it.

Reply:
Note: I am wondering if the custom interpolator feature of SG can be compatible with VFX?
https://github.com/Unity-Technologies/Graphics/pull/3281/files#diff-a1d073cb0296b152768b86b960e65e0a456db463b2ec858f7bbcc5a5c0fe675eR35

what do you think, can we simply add CustomInterpolators customInterpolators in the fraginputs of VFX?

Reply:
>Not very happy with how I currently move the entire struct splice up to this point, eager to hear thoughts about it.
I have for now no better idea, maybe we can revisit this in a second time, but gess that for now it is fine like this

Reply:
Late reply on my thoughts to this- but overall I think it is definitely possible to utilize custom interpolators to transfer the VFX properties (it would also solve this procedural FragInputs since the way custom interpolators handle it, the just have a substruct wrapped in a preprocessor). 

But, I think it is out of the scope of this PR to do it. As you say, a second time/PR seems correct for this